src/movies.js

```diff
@@ -1,25 +1,85 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+function getAllDirectors(moviesArray) {
+    const justDirectors = moviesArray.map((movies) => movies.director);
+    return justDirectors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    const dramaMovies = moviesArray.filter((movies) => movies.director === "Steven Spielberg" && movies.genre.includes("Drama"));
+        return dramaMovies.length;
+    }
+    
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+
+    const scores = moviesArray.reduce((acc, val) => {
+        if(val.score){
+            return acc + val.score;     
+        } else{
+            return acc;
+        }
+  }, 0);  
+
+   return Math.round ((scores / moviesArray.length) * 100) / 100;
+   
+}
+   
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+    const dramaScores = moviesArray.reduce((acc, val) => {
+        if(val.genre.includes("Drama") && val.score){
+            return acc + val.score;     
+        } else{
+            return acc;
+        }
+  }, 0);  
+
+   return Math.round ((dramaScores / moviesArray.length) * 100) / 100;
+   
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
```

Comment:
This iteration is not passing this test `"should order movies with the same year by their title, alphabetically"`
In your last condition you have (a.title < b.title) it is correct but, first you need to check if a.year is equal to b.year, and then if this is true you need to compare the titles to decide which one comes first:

```
function orderByYear(moviesArray) {
  const orderedByYear = moviesArray.slice();
  function compare(a, b) {
    if (a.year > b.year) {
      return 1;
    } else if (a.year < b.year) {
      return -1;

    } else if (a.year === b.year);
    return a.title.localeCompare(b.title);
  }

  return orderedByYear.sort(compare);
}
```